### PR TITLE
Feat(ticket#257): Create leave server action

### PR DESCRIPTION
**Pull request by ghulamhafeez**
## Change Request

- Create a server action to get all leave records and filter them to display the count of approved leaves in a column
- Also improve scenario & some more minor things

### Resources
image

### CODE REVIEW

File: src/app/(app)/attendance-report/actions/attendance-report.ts [hunk 1]
```diff
@@ -27,67 +30,50 @@ export const getAttendanceReport = async () => {
...
-        totalAttendanceDays(attendance.created_at)
-    );
+    const memberAttendance = attendanceData.filter(({ memberId: id }) => id === memberId);
 
     const presentDays = new Set(
       memberAttendance
-        .filter((attendance) => attendance.status === AttendanceStatus.Approve)
-        .map((attendance) => attendance.created_at?.split("T")[0])
+        .filter(({ status }) => status === AttendanceStatus.Approve)
+        .map(({ created_at }) => created_at.split("T")[0])
```
Comment (nadirhussain786) [inline src/app/(app)/attendance-report/actions/attendance-report.ts:48]: @ghulamhafeez I think we should make use of the date function.
```suggestion
        .map(({ created_at }) => created_at.toLocaleDateString()
```

File: src/app/(app)/leaves/actions/get-leave-request-by-date.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Tables } from "@/lib/db";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+
+export const getleaves = async () => {
+  const supabase = await getSupabaseClient();
+
+  // Get the current month's start and end dates
+  const now = new Date();
+  const startOfMonth = new Date(now.getFullYear(), now.getMonth(), 1).toISOString();
+  const endOfMonth = new Date(now.getFullYear(), now.getMonth() + 1, 0, 23, 59, 59, 999).toISOString();
```
Comment (nadirhussain786) [inline src/app/(app)/leaves/actions/get-leave-request-by-date.ts:10]: @ghulamhafeez  instead of doing these calculation we should use `moment` library

**Review by muhammad-bilal-ali-khokhar — APPROVED** [review]

**Review by Mubashar1009 — APPROVED** [review]

File: src/app/(app)/leaves/actions/get-leave-request-by-date.ts (hunk 2 above)
Comment (ghulamhafeez) [reply src/app/(app)/leaves/actions/get-leave-request-by-date.ts]: Fixed

File: src/app/(app)/attendance-report/actions/attendance-report.ts (hunk 1 above)
Comment (ghulamhafeez) [reply src/app/(app)/attendance-report/actions/attendance-report.ts]: Fixed

**Review by muhammad-bilal-ali-khokhar — APPROVED** [review]

**Pull request merged**